content/blog/2022-05-19-fast-brili.md

```diff
@@ -0,0 +1,141 @@
++++
+title = "Bril bytecode interpreter"
+[extra]
+bio = """
+	Charles Sherk is an junior studying Computer Science & Mathematics at
+	Cornell university, interested in compilers and programming language
+	implementation.
+	Susan Garry is a senior studying Computer Science at Cornell University,
+	interested in compilers and programming language design.
+
+"""
+[[extra.authors]]
+name = "Charles Sherk"
+link = "https://github.com/charles-rs/"
+[[extra.authors]]
+name = "Susan Garry"
+link = "https://github.com/susan-garry"
++++
+
+## Motivation
+Currently we have access to `brili`, which is written with an emphasis on
+readable code and extensiblity, and `brili-rs`, which is supposed to be faster,
+but `brili-rs` isn't actually that fast, and seems to be rather bloated for what
```

Comment:
"Bloated" seems a little pejorative! I think it's just pretty general and meant as a showcase for `bril-rs`, the underlying library that lets people do their homework in Rust. It's, like, an interpreter written in Rust, which happens to be a fast language—not an interpreter hyper-optimized for speed.

Reply:
fair enough, it's presented [here](https://capra.cs.cornell.edu/bril/tools/brilirs.html) as "fast interpreter" and i may have read a little much into that lol